reflect.go

```diff
@@ -106,6 +130,141 @@ func inspect(t reflect.Type, jsonTag string) Property {
 	return p
 }
 
+// buildMapType build map type swag info
+func buildMapType(mapType reflect.Type) *AdditionalProperties {
+
+	prop := AdditionalProperties{}
+
+	// get the element object of the map
+	// Example:
+	//	mapType ==> map[string][]*Struct|Struct|*string|string
+	//  or mapTYpe ==> map[string]*Struct|Struct|*string|string
+	//	mapType.Elem().Kind() ==> []*Struct|Struct|*string|string
+	//  or mapType.Elem().Kind() ==> *Struct|Struct|*string|string
+	if mapType.Elem().Kind().String() != "interface" {
+		isSlice := isSliceOrArryType(mapType.Elem().Kind())
+		if isSlice || isByteArrayType(mapType.Elem()) {
+			// if map value is slice
+			// Example:
+			//   mapType.Elem()==> []*Struct|Struct|*string|string
+			mapType = mapType.Elem()
+		}
+
+		// if map value is struct or built-in primitive type
+		// Example:
+		//    mapType.Elem()==> *Struct|Struct|*string|string
+		isPrimitive := isPrimitiveType(mapType.Elem().Name(), mapType.Elem())
+
+		if isByteArrayType(mapType.Elem()) {
+			prop.Type = "string"
+		} else {
+			if isSlice {
+				prop.Type = types.Array.String()
+				prop.Items = &Items{}
+				if isPrimitive {
+					prop.Items.Type = jsonSchemaType(mapType.Elem().String(), mapType.Elem())
+					prop.GoType = getGoType(mapType.Elem())
+				} else {
+					prop.Items.Ref = makeMapRef(mapType.Elem().String())
+					prop.GoType = getGoType(mapType.Elem())
+				}
+			} else if isPrimitive {
+				prop.Type = jsonSchemaType(mapType.Elem().String(), mapType.Elem())
+				prop.GoType = getGoType(mapType.Elem())
+			} else {
+				prop.Ref = makeMapRef(mapType.Elem().String())
+				prop.GoType = getGoType(mapType.Elem())
+			}
+		}
+	}
+
+	return &prop
+}
+
+func getGoType(t reflect.Type) reflect.Type {
+
+	var goType reflect.Type
+
+	if t.Kind() == reflect.Ptr {
+		goType = t.Elem()
+	} else {
+		goType = t
+	}
+
+	return goType
+}
+
+func makeMapRef(typeName string) string {
+	type_name := strings.Trim(typeName, "*")
+	return makeRef(type_name)
+}
+
+func isSliceOrArryType(t reflect.Kind) bool {
+	return t == reflect.Slice || t == reflect.Array
+}
+
+func isByteArrayType(t reflect.Type) bool {
+	return (t.Kind() == reflect.Slice || t.Kind() == reflect.Array) &&
+		t.Elem().Kind() == reflect.Uint8
+}
```

Comment:
Why need to check `uint8` ?

Reply:
the code checks whether the given type represents a byte slice type, with uint8 confirming that the elements within the slice or array are bytes.

Reply:
I know what this code means, I just don't understand why `[]uint8` needs to be judged separately.

Reply:
where would it be more reasonable for me to check if it's a byte slice?

Reply:
Maybe I didn’t express it clearly, please take a look at the review suggestion below.